source/input.cpp

```diff
@@ -738,3 +831,777 @@ extern "C" BOOL WINAPI HookGetCursorPosition(LPPOINT lpPoint)
+		{
+			bool remove = false;
+			
+			switch (pItem->header.dwType)
+			{
+				case RIM_TYPEKEYBOARD:
+					remove = block_keyboard;
+					break;
+				case RIM_TYPEMOUSE:
+					remove = block_mouse;
+					break;
+				// Gamepad, ReShade does not care about this...?
+				default:
+					remove = block_gamepad;
+					break;
+			}
+
+			// If item is not removed, append it to the buffer of RAWINPUT
+			//   packets we are allowing the game to see
+			if (remove == false)
+			{
+				memcpy (pOutput, pItem, pItem->header.dwSize);
+				        pOutput = NEXTRAWINPUTBLOCK (pOutput);
+
+				++count;
+			}
+
+			else
+			{
+				bool keyboard = pItem->header.dwType == RIM_TYPEKEYBOARD;
+				bool mouse    = pItem->header.dwType == RIM_TYPEMOUSE;
+
+				// Clearing all bytes above would have set the type to mouse, and some games
+				//   will actually read data coming from RawInput even when the size returned is 0!
+				pItem->header.dwType = keyboard ? RIM_TYPEKEYBOARD :
+				                          mouse ? RIM_TYPEMOUSE    :
+				                                  RIM_TYPEHID;
+
+				// Supplying an invalid device will early-out SDL before it calls HID APIs to try
+				//   and get an input report that we don't want it to see...
+				pItem->header.hDevice = nullptr;
+
+				// Most engines will honor the Input Sink state
+				pItem->header.wParam  = RIM_INPUTSINK;
+
+				if (keyboard)
+				{
+					if (! (pItem->data.keyboard.Flags & RI_KEY_BREAK))
+					       pItem->data.keyboard.VKey  = 0;
+
+					// Fake key release
+					pItem->data.keyboard.Flags |= RI_KEY_BREAK;
+				}
+
+				// Block mouse input by zeroing-out the memory; not all games will
+				//   see *pcbSize=0 and RIM_INPUTSINK and ignore input...
+				else
+				{
+					memset (&pItem->data.mouse, 0,
+					         pItem->header.dwSize - sizeof (RAWINPUTHEADER));
+				}
+
+				memcpy (pOutput, pItem, pItem->header.dwSize);
+				        pOutput = NEXTRAWINPUTBLOCK (pOutput);
+
+				++count;
+			}
+
+			pItem = NEXTRAWINPUTBLOCK (pItem);
+		}
+		
+		*pcbSize = (UINT)((uintptr_t)pOutput - (uintptr_t)pData);
+		
+		return count;
+	}
+
+	return trampoline(pData, pcbSize, cbSizeHeader);
+}
+
+
+
+#include <concurrent_unordered_map.h>
+
+class ReShade_Win32_WindowHookManager {
+public:
+  concurrency::concurrent_unordered_map <
+    DWORD, HOOKPROC > _RealMouseProcs;
+           HOOKPROC   _RealMouseProc    = nullptr;
+  concurrency::concurrent_unordered_map <
+    DWORD, HHOOK >    _RealMouseHooks;
+           HHOOK      _RealMouseHook    = nullptr;
+
+  concurrency::concurrent_unordered_map <
+    DWORD, HOOKPROC > _RealKeyboardProcs;
+           HOOKPROC   _RealKeyboardProc = nullptr;
+  concurrency::concurrent_unordered_map <
+    DWORD, HHOOK >    _RealKeyboardHooks;
+           HHOOK      _RealKeyboardHook = nullptr;
+} __hooks;
+
+//
+// Some typedefs because I cannot figure out what the requirements to actually
+// get this to build are, WinUser.h seems gimped by something :)
```

Comment:
This is because of https://github.com/crosire/reshade/blob/main/deps/Windows.props#L7, which reduces the amount of stuff pulled in via Windows headers. I think `NOWH` is the culprit specifically here, so that can just be removed from the props file and all the definitions should appear.